pkg/network/deviceinfo/deviceinfo_test.go

```diff
@@ -0,0 +1,208 @@
+	)
+
+	networkStatusWithMixedNetworksFmt := `
+	[
+	{
+	"name": "kindnet",
+	"interface": "eth0",
+	"ips": [
+	  "10.244.1.9"
+	],
+	"mac": "3a:7e:42:fa:37:c6",
+	"default": true,
+	"dns": {}
+	},
+	{
+	"name": "default/nad1",
+	"interface": "%s",
+	"mac": "8a:37:d9:e7:0f:18",
+	"dns": {}
+	},
+	{
+	"name": "default/nad2",
+	"interface": "%s",
+	"dns": {},
+	"device-info": {
+	  "type": "pci",
+	  "version": "1.0.0",
+	  "pci": {
+	    "pci-address": "0000:65:00.2"
+	  }
+	}
+	}
+	]`
+	networkStatusWithNoDeviceInfoFmt := `
+[
+{
+ "name": "kindnet",
+ "interface": "eth0",
+ "ips": [
+   "10.244.2.131"
+ ],
+ "mac": "82:cf:7c:98:43:7e",
+ "default": true,
+ "dns": {}
+},
+{
+ "name": "default/nad1",
+ "interface": "%s",
+ "dns": {},
+}
+]`
+	networkStatusWithPrimaryInterfaceOnlyFmt := `
+[
+{
+ "name": "kindnet",
+ "interface": "eth0",
+ "ips": [
+   "10.244.2.131"
+ ],
+ "mac": "82:cf:7c:98:43:7e",
+ "default": true,
+ "dns": {}
+}
+]`
+	DescribeTable("should prepare empty network device info annotation",
+		func(networkList []virtv1.Network, interfaceList []virtv1.Interface, networkStatusAnnotationValue string) {
+			networkDeviceInfoAnnotationValue := deviceinfo.CreateNetworkDeviceInfoAnnotationValue(networkList, interfaceList, networkStatusAnnotationValue)
+			Expect(networkDeviceInfoAnnotationValue).To(Equal(""))
+		},
+		Entry("when networkStatusAnnotation deson't contain the interface",
+			[]virtv1.Network{
+				newMultusNetwork("foo", "default/nad1"),
+			},
+			[]virtv1.Interface{
+				newBindingPluginInterface("foo"),
+			},
+			fmt.Sprintf(networkStatusWithPrimaryInterfaceOnlyFmt),
+		),
+		Entry("when networkStatusAnnotation has no interfaces with device info",
+			[]virtv1.Network{
+				newMultusNetwork("foo", "default/nad1"),
+			},
+			[]virtv1.Interface{
+				newBindingPluginInterface("foo"),
+			},
+			fmt.Sprintf(networkStatusWithNoDeviceInfoFmt, fooHashedIfaceName, booHashedIfaceName),
+		),
+		Entry("when pod's networkStatus Annotation does not exist",
+			[]virtv1.Network{newMultusNetwork("foo", "default/nad1")},
+			[]virtv1.Interface{newBindingPluginInterface("foo")},
+			"",
+		),
+	)
+
+	DescribeTable("should prepare non empty network device info annotation",
+		func(networkList []virtv1.Network, interfaceList []virtv1.Interface, networkStatusAnnotationValue, expectedAnnotation string) {
+			Expect(deviceinfo.CreateNetworkDeviceInfoAnnotationValue(networkList, interfaceList, networkStatusAnnotationValue)).To(Equal(expectedAnnotation))
+		},
+
+		Entry("when given Interfaces{1X primary- no device info, 1X no device info, 1X with device info, 1X not in multus status}",
+			[]virtv1.Network{
+				newMasqueradeDefaultNetwork(),
+				newMultusNetwork("boo", "default/nad1"),
+				newMultusNetwork("foo", "default/nad2"),
+				newMultusNetwork("doo", "default/nad3"),
+			},
+			[]virtv1.Interface{
+				newMasqueradePrimaryInterface(),
+				newBridgeInterface("boo"), newBindingPluginInterface("foo"),
```

Comment:
It seems network and interfaces represent the VMI spec and they should have 1:1 relation, but `doo` interface is missing.
nit: missing new line for ` newBindingPluginInterface`.